src/Shared/Http2cat/Http2ManualFrameWriter.cs

```diff
@@ -0,0 +1,560 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Pipelines;
+using System.Net.Http.HPack;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
+
+namespace Microsoft.AspNetCore.Http2Cat
+{
+    internal class Http2ManualFrameWriter
```

Comment:
I _really_ don't like that this isn't shared with the Http2FrameWriter, considering around 90% of it is the same.

Reply:
Can we do partials to add in functionality necessary in Kestrel?

Reply:
This is where I had to draw the line. The Frame types and parsers didn't have any dependencies, but Http2FrameWriter pulls in the entire Kestrel object graph of things we didn't want/need (flow control, streams, header structures, etc.. You're right, removing those dependencies removed about 10-20% of the code here.

Reply:
Removed. It turns out I only needed two of the static APIs.